op-challenger/game/fault/trace/vm/kona_vm_config.go

```diff
@@ -0,0 +1,51 @@
+package vm
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/trace/utils"
+	"github.com/ethereum-optimism/optimism/op-node/chaincfg"
+)
+
+type KonaVmConfig struct {
+	Config
+}
+
+var _ VmConfig = (*KonaVmConfig)(nil)
+
+func NewKonaVmConfig(vmConfig Config) *KonaVmConfig {
+	return &KonaVmConfig{
+		vmConfig,
+	}
+}
+
+func (s *KonaVmConfig) Cfg() Config {
+	return s.Config
+}
+
+func (s *KonaVmConfig) FillHostCommand(args []string, dataDir string, inputs utils.LocalGameInputs) ([]string, error) {
+	if args == nil {
+		return nil, errors.New("args is nil")
+	}
+	if s.Network == "" {
+		return nil, errors.New("Network is not defined")
+	}
```

Comment:
Missed this the first time, but this isn't going to work - network is an optional param and only works for networks in the super chain registry.  In e2e tests and many other places we specify the network by providing the rollup config and genesis directly, leaving network unset.

As a temporary work around you could load the chain ID from the rollup config if network isn't set, but kona is almost certainly going to fail in that case because the rollup config probably isn't in the super chain registry and needs to be specified manually. Won't be hit in production but is required for compatibility with things like the local devnet or e2e tests.

Reply:
We'll need to update kona to support this. This will need to be a follow-up pr if we'd like to add tests to the devnet/e2e